pkg/services/rpcsrv/server.go

```diff
@@ -2759,6 +2759,12 @@ func (s *Server) subscribe(reqParams params.Params, sub *subscriber) (any, *neor
 			return nil, neorpc.WrapErrorWithData(neorpc.ErrInvalidParams, err.Error())
 		}
 	}
+	if filter != nil {
+		err = filter.IsValid()
+		if err != nil {
+			return nil, neorpc.WrapErrorWithData(neorpc.ErrInvalidParams, err.Error())
+		}
+	}
```

Comment:
Good, we need to test this server-side functionality.

Reply:
Create a separate test in `services/rpcsrv/subscription_test.go`, it may have a structure similar to `TestFilteredSubscriptions`, but we only need to check failed/successful response from `subscribe` RPC request.